### PR TITLE
Cleanup implementation of Bitcoin amounts

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: comit/src/network/swap_digest.rs [hunk 1]
```diff
@@ -88,6 +88,10 @@ pub struct HbitHerc20 {
     pub token_contract: identity::Ethereum,
 }
 
+/// A new-type that allows us to implemented `ToDigestInput` on foreign types.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Digestable<T>(pub T);
```
Comment (D4nte) [inline comit/src/network/swap_digest.rs:93]: Does the inner need to be pub? Maybe better to implement `From`.

File: nectar/src/order.rs [hunk 2]
```diff
@@ -66,7 +66,7 @@ impl BtcDaiOrderForm {
 
         Ok(BtcDaiOrderForm {
             position: Position::Sell,
-            quantity: base_amount.into(),
+            quantity: Quantity::new(base_amount),
```
Comment (D4nte) [inline nectar/src/order.rs:69]: Is it a sign that `From` implementation should be done?

File: nectar/src/order.rs [hunk 3]
```diff
@@ -225,7 +225,7 @@ mod tests {
         )
         .unwrap();
 
-        assert_eq!(bitcoin::Amount::from(order.quantity), btc(8.0));
+        assert_eq!(order.quantity.to_inner(), btc(8.0));
```
Comment (D4nte) [inline nectar/src/order.rs:228]: Is it a sign that From implementation should be done?

**Review by D4nte — APPROVED** [review]
Noice! 👏

File: comit/src/network/swap_digest.rs (hunk 1 above)
Comment (thomaseizinger) [reply comit/src/network/swap_digest.rs:93]: This struct doesn't enforce any invariants among its inner field so I don't see a reason why it should be private. Adding a `From` impl is just more code for no benefit IMO.

File: nectar/src/order.rs (hunk 2 above)
Comment (thomaseizinger) [reply nectar/src/order.rs:69]: I am not sure to be honest.

I am not abstracting over the construction of `Quantity` anywhere so I don't see anything wrong with a using an explicit constructor. For a type like `Quantity` that enforces invariants, I actually prefer it being explicit so it stands out among all the other `.into()` calls. Can't really present any better argument though :)

Comment (tcharding) [reply nectar/src/order.rs:69]: I'm with @D4nte on this one, although I don't feel strongly, I prefer the from/into version than new/to_inner.

**Review by tcharding — APPROVED** [review]

**Pull request merged**